### PR TITLE
Fix broadcasting of messages with headers with NULL or all-whitespace values

### PR DESCRIPTION
**Pull request by habbbe**
Fix parsing of headers with NULL or all-whitespace values when lifting

Add tests for NULL and all-whitespace header values

### CODE REVIEW

File: test/test.c [hunk 1]
```diff
@@ -20109,6 +20109,12 @@ void test_HeadersBasic(void)
     IFOK(s, natsMsgHeader_Set(msg, "Headers", "Hello Headers!"))
     testCond(s == NATS_OK);
 
+    IFOK(s, natsMsgHeader_Add(msg, "NULL header", NULL))
+    testCond(s == NATS_OK);
+
+    IFOK(s, natsMsgHeader_Add(msg, "Whitespace header", "  "))
+    testCond(s == NATS_OK);
+
```
Comment (kozlovic) [inline test/test.c:20117]: Let's add some `test()` so that when running the test in verbose, we see what we are testing and the `PASSED` result. The name of the keys are wrong. They need to be of the form `This-Is-Correct-Name`. The last part I suggest adding is to make sure that your changes still work when adding more than one of those "bad" values to the same key, and later retrieve them with `Values()` API:
```
    test("Add NULL header: ");
    IFOK(s, natsMsgHeader_Add(msg, "Null-Header", NULL))
    testCond(s == NATS_OK);

    test("Add whitespace header: ");
    IFOK(s, natsMsgHeader_Add(msg, "Whitespace-Header", "  "))
    testCond(s == NATS_OK);

    test("Add NULL and whitespace values under same header: ");
    IFOK(s, natsMsgHeader_Add(msg, "Special-Headers", NULL))
    IFOK(s, natsMsgHeader_Add(msg, "Special-Headers", "  "))
    testCond(s == NATS_OK);
```

**Review by kozlovic — CHANGES_REQUESTED** [review]
I was going to say that we should simply fix the fact that we allow NULL or empty values, but I checked the Go client, and those are not rejected. Even worse, on the Go client, there is no error returned when a key is not present, we just return "", so user will not really know if the key was present but empty or the value is empty. 
So I am willing to accept those changes. I have however made some comments on the tests.

**Comment by habbbe** [comment]
Added suggestions. I also added an additional test for completely empty value.

**Review by kozlovic — APPROVED** [review]
LGTM - Thank you for your contribution!
